### PR TITLE
pdf renderer and tag cloud

### PR DESCRIPTION
**Pull request by emswbandara**

### CODE REVIEW

File: modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java [hunk 1]
```diff
@@ -1241,6 +1247,152 @@ public String uploadDocument(String processName, String processVersion, String d
...
+							processXML.getElementsByTagName("version").item(0).getTextContent();
+
+					JSONObject processJSON = new JSONObject();
+					processJSON.put("path", processPath);
+					processJSON.put("processid", processResource.getUUID());
+					processJSON.put("processname", processName);
+					processJSON.put("processversion", processVersion);
+					Tag[] tags = reg.getTags(processPath);
+
+					for(Tag tag : tags){
```
Comment (madurangasiriwardena) [inline modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java:1281]: Formatting issue

File: modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java [hunk 2]
```diff
@@ -1241,6 +1247,152 @@ public String uploadDocument(String processName, String processVersion, String d
...
+
+						Iterator<String> keys = tagsObj.keys();
+
+						if(!keys.hasNext()  || keys == null){
```
Comment (madurangasiriwardena) [inline modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java:1285]: Formatting issue

File: modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java [hunk 3]
```diff
@@ -1241,6 +1247,152 @@ public String uploadDocument(String processName, String processVersion, String d
...
+							JSONArray newTagArray = new JSONArray();
+							newTagArray.put(processJSON);
+							tagsObj.put(tag.getTagName(), newTagArray);
+							continue;
+						}
+
+
+						while(keys.hasNext()){
```
Comment (madurangasiriwardena) [inline modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java:1293]: Formatting issue

File: modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java [hunk 4]
```diff
@@ -1241,6 +1247,152 @@ public String uploadDocument(String processName, String processVersion, String d
...
+							String temp = (keys.next());
+
+							if(temp == tag.getTagName()){
```
Comment (madurangasiriwardena) [inline modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java:1296]: Formatting issue

File: modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java [hunk 5]
```diff
@@ -1241,6 +1247,152 @@ public String uploadDocument(String processName, String processVersion, String d
...
+								JSONArray processArray = ((JSONArray) tagsObj.get(temp));
+								processArray = processArray.put(processJSON);
+								tagsObj.put(temp, processArray);
+								break;
+
+							}
+							if(!keys.hasNext()){
```
Comment (madurangasiriwardena) [inline modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java:1303]: Formatting issue

File: modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java (hunk 1 above)
Comment (emswbandara) [reply modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java]: Fixed with 0a2b761b79658813ff2c081e3eca521037de97a2

File: modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java (hunk 2 above)
Comment (emswbandara) [reply modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java]: Fixed with 0a2b761

File: modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java (hunk 1 above)
Comment (emswbandara) [reply modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java:1281]: Fixed with 044fdb57fa82948dbc941554d9a41096e7104533

File: modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java (hunk 2 above)
Comment (emswbandara) [reply modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java:1285]: Fixed with 044fdb57fa82948dbc941554d9a41096e7104533

File: modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java (hunk 3 above)
Comment (emswbandara) [reply modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java:1293]: Fixed with 044fdb57fa82948dbc941554d9a41096e7104533

File: modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java (hunk 4 above)
Comment (emswbandara) [reply modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java:1296]: Fixed with 044fdb57fa82948dbc941554d9a41096e7104533

File: modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java (hunk 5 above)
Comment (emswbandara) [reply modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java:1303]: Fixed with 044fdb57fa82948dbc941554d9a41096e7104533

**Pull request merged**